### ISSUE TITLE
SSL using PEM files on macos

### ISSUE THREAD

**Issue by ppatt** [blackbeam/rust-mysql-simple#140]
Hi

 Is there a reason behind SSL not being supported with PEM files on macOS? Are there some limitations of the macOS openssl compared to Linux?

Thanks

**Comment by blackbeam** [comment]
Hi.

It's not obvious how to use PEM with security_framework API. The [only option][1] that I found is importing PKCS#12 data.

Btw creating PKCS#12 container out of your PEM is quite easy:

```sh
$ openssl pkcs12 -export -inkey cert_key_pem.txt -in cert_key_pem.txt -out cert_key.p12
```

[1]: http://sfackler.github.io/rust-security-framework/doc/v0.2/security_framework/import_export/struct.Pkcs12ImportOptions.html#method.import

**Comment by ppatt** [comment]
Thanks for the response. Using PKCS12 does not work for me, because I don't have access to the private key. However I could connect with macos using openssl. 
Other users who'd like to connect to amazon RDS from a mac are in the same situation as me. I could create a PR with some changes to enable openssl for mac if you're open for it.

**Comment by ppatt** [comment]
Hi

So I added an open-ssl feature to use openssl instead of security-framework on mac. 
This is the PR: https://github.com/blackbeam/rust-mysql-simple/pull/141, unfortunately I couldn't make it so that the dependency is feature based, as I saw from the documentation this is not possible.

I guess you had a reason to use security-framework over openssl on mac, so I'll just post this here for people who run into this same problem and wouldn't like to switch to openssl by themselves.
